### DISCUSSION TITLE
Lazy middleware (load after page load)

### DISCUSSION THREAD

**Discussion by cosbgn** [nuxt/nuxt#21234] [General]
I have a middleware which checks how many credits a user have. The middleware will never redirect the user anywhere and is not used immediately. Something like this:

```js
// middleware/billing.js
export default defineNuxtRouteMiddleware(async (to, from) => {
	if (process.server){ return; }
	const billing = useState('billing')
	billing.value = billing.value ?? await $fetch("/api/billing", {lazy:true, server:false})
})
```

Is there a way to running it lazily after page load? Something like `onMounted`? Currently it blocks navigation and slows down first page load by a second or so.

**Comment by warflash** [comment]
What's the reason for putting this inside a Middleware and not something like pinia or a regular composable?

**Reply by cosbgn** [reply]
That's actually a good idea. This works fine:

```js
// composables/states.js

export function useBilling() {
	const billing = useState('billing')
	onMounted(async () => {
		billing.value = billing.value ?? await $fetch("/api/billing", {lazy:true, server:false})
	})
	return billing
}
```
